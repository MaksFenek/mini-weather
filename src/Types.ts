export interface IWeatherData {
  weather: [
    {
      main: string;
      description: string;
    }
  ];
  main: {
    temp: number;
    pressure: number;
    humidity: number;
  };
  wind: {
    speed: number;
    deg: number;
  };
  coord: {
    lon: number;
    lat: number;
  };
}

export interface IDay {
  name: string;
  temp: {
    day: number;
    night: number;
  };
  pressure: number;
  humidity: number;
  weather: [
    {
      main: string;
    }
  ];
  wind_speed: number;
  wind_deg: number;
}

export interface IForecastData {
  daily: IDay[];
}
