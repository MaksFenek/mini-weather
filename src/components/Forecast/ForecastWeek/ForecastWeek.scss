.week {
  width: 100%;
  grid-area: week;
  display: flex;
  margin-top: 14px;
  overflow-x: scroll;
}
