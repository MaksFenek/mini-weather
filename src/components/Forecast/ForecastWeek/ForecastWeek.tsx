// React
import React from 'react';

// Styles
import './ForecastWeek.scss';

// Types
import { IDay } from '../../../Types';
import Card from './Card/Card';
export interface IForecastWeek {
  week: IDay[] | undefined;
}

const ForecastWeek: React.FC<IForecastWeek> = ({ week }) => {
  return (
    <div aria-label='week' className='week'>
      {week &&
        week.map((day, index) => {
          if (index < 7) return <Card key={index} {...day} />;
        })}
    </div>
  );
};

export default ForecastWeek;
