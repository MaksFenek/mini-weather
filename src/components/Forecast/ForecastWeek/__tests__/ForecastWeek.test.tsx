import React from 'react';
import ForecastWeek from '../ForecastWeek';
import { mount } from 'enzyme';
import { IDay } from '../../../../Types';
import renderer from 'react-test-renderer';

describe('<ForecastWeek />', () => {
  const week: IDay[] = [
    {
      name: 'Monday',
      temp: {
        day: 13,
        night: 13,
      },
      pressure: 13,
      humidity: 13,
      weather: [
        {
          main: 'clouds',
        },
      ],
      wind_speed: 13,
      wind_deg: 13,
    },
    {
      name: 'Tuesday',
      temp: {
        day: 13,
        night: 13,
      },
      pressure: 13,
      humidity: 13,
      weather: [
        {
          main: 'clear',
        },
      ],
      wind_speed: 13,
      wind_deg: 13,
    },
  ];

  const wrapper = mount(<ForecastWeek week={...week} />);
  const tree = renderer.create(<ForecastWeek week={...week} />).toJSON();
  it('contains all own children', () => {
    expect(wrapper.find(`[aria-label='week']`).children()).toHaveLength(2);
    expect(tree).toMatchSnapshot();
  });
});
