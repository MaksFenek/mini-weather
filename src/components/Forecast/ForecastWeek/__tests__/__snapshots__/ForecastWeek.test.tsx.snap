// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ForecastWeek /> contains all own children 1`] = `
<div
  aria-label="week"
  className="week"
>
  <div
    className="card"
  >
    <div
      aria-label="name"
      className="card-day"
    >
      Monday
    </div>
    <svg
      className="card-icon"
      height="84px"
      width="84px"
    >
      <use
        aria-label="icon"
        xlinkHref="icons/icons.svg#clouds"
      />
    </svg>
    <div
      className="card-weather"
    >
      <span
        aria-label="temp-day"
      >
        13
      </span>
       째C /
       
      <span
        aria-label="temp-night"
      >
        13
      </span>
       째C
    </div>
    <ul
      className="card-params"
    >
      <li>
        <svg
          className="icon"
          height="36px"
          width="36px"
        >
          <use
            xlinkHref="icons/params.svg#humidity"
          />
        </svg>
        <p>
          <span
            aria-label="humidity"
          >
            13
          </span>
           %
        </p>
      </li>
      <li>
        <svg
          className="icon"
          height="36px"
          width="36px"
        >
          <use
            xlinkHref="icons/params.svg#barometer"
          />
        </svg>
        <p>
          <span
            aria-label="pressure"
          >
            13
          </span>
           hPa
        </p>
      </li>
      <li>
        <svg
          className="icon"
          height="52px"
          style={
            Object {
              "margin": "-7px",
            }
          }
          width="52px"
        >
          <use
            xlinkHref="icons/compass.svg#main
              "
          />
          <use
            style={
              Object {
                "transform": "rotate(283deg)",
                "transformOrigin": "center",
              }
            }
            xlinkHref="icons/compass.svg#arrow"
          />
        </svg>
        <p>
          <span
            aria-label="wind_speed"
          >
            13
          </span>
           
          mph
        </p>
      </li>
    </ul>
  </div>
  <div
    className="card"
  >
    <div
      aria-label="name"
      className="card-day"
    >
      Tuesday
    </div>
    <svg
      className="card-icon"
      height="84px"
      width="84px"
    >
      <use
        aria-label="icon"
        xlinkHref="icons/icons.svg#clear"
      />
    </svg>
    <div
      className="card-weather"
    >
      <span
        aria-label="temp-day"
      >
        13
      </span>
       째C /
       
      <span
        aria-label="temp-night"
      >
        13
      </span>
       째C
    </div>
    <ul
      className="card-params"
    >
      <li>
        <svg
          className="icon"
          height="36px"
          width="36px"
        >
          <use
            xlinkHref="icons/params.svg#humidity"
          />
        </svg>
        <p>
          <span
            aria-label="humidity"
          >
            13
          </span>
           %
        </p>
      </li>
      <li>
        <svg
          className="icon"
          height="36px"
          width="36px"
        >
          <use
            xlinkHref="icons/params.svg#barometer"
          />
        </svg>
        <p>
          <span
            aria-label="pressure"
          >
            13
          </span>
           hPa
        </p>
      </li>
      <li>
        <svg
          className="icon"
          height="52px"
          style={
            Object {
              "margin": "-7px",
            }
          }
          width="52px"
        >
          <use
            xlinkHref="icons/compass.svg#main
              "
          />
          <use
            style={
              Object {
                "transform": "rotate(283deg)",
                "transformOrigin": "center",
              }
            }
            xlinkHref="icons/compass.svg#arrow"
          />
        </svg>
        <p>
          <span
            aria-label="wind_speed"
          >
            13
          </span>
           
          mph
        </p>
      </li>
    </ul>
  </div>
</div>
`;
